arch/arm/core/cortex_m/mpu/nxp_mpu.c

```diff
@@ -97,82 +110,256 @@ static void _region_init(u32_t index, u32_t region_base,
+				(!_mpu_partition_is_sane(&regions[i]))) {
+			__ASSERT(0, "Partition %u: sanity check failed.", i);
+			return -EINVAL;
+		}
+
+#if defined(CONFIG_MPU_STACK_GUARD)
+		if (regions[i].attr.ap_attr == MPU_REGION_SU_RX) {
+
+			/* Attempt to configure an MPU Stack Guard region; this
+			 * will require splitting of the underlying SRAM region
+			 * into two SRAM regions, leaving out the guard area to
+			 * be programmed afterwards.
+			 */
+			reg_index =
+				_mpu_sram_partitioning(reg_index, &regions[i]);
+		}
+#endif /* CONFIG_MPU_STACK_GUARD */
+
+		if (reg_index == -EINVAL) {
+			return reg_index;
+		}
+
+		reg_index = _mpu_configure_region(reg_index, &regions[i]);
+
+		if (reg_index == -EINVAL) {
+			return reg_index;
+		}
+
+		/* Increment number of programmed MPU indices. */
+		reg_index++;
+	}
+
+	return reg_index;
+}
+
+/* This internal function programs the static MPU regions.
+ *
+ * It returns the number of MPU region indices configured.
+ *
+ * Note:
+ * If the static MPU regions configuration has not been successfully
+ * performed, the error signal is propagated to the caller of the function.
+ */
+static int _mpu_configure_static_mpu_regions(const struct k_mem_partition
+	static_regions[], const u8_t regions_num,
+	const u32_t background_area_base,
+	const u32_t background_area_end)
+{
+	int mpu_reg_index = static_regions_num;
+
+	/* In NXP MPU architecture the static regions are
+	 * programmed on top of SRAM region configuration.
+	 */
+	ARG_UNUSED(background_area_base);
+	ARG_UNUSED(background_area_end);
+
+	mpu_reg_index = _mpu_configure_regions(static_regions,
+		regions_num, mpu_reg_index, true);
+
+	static_regions_num = mpu_reg_index;
+
+	return mpu_reg_index;
+}
+
+/* This internal function programs the dynamic MPU regions.
+ *
+ * It returns the number of MPU region indices configured.
+ *
+ * Note:
+ * If the dynamic MPU regions configuration has not been successfully
+ * performed, the error signal is propagated to the caller of the function.
+ */
+static int _mpu_configure_dynamic_mpu_regions(const struct k_mem_partition
+		dynamic_regions[], u8_t regions_num)
+{
+	/* Reset MPU regions inside which dynamic memory regions may
+	 * be programmed.
+	 */
+	_region_init(mpu_config.sram_region, (const struct nxp_mpu_region *)
+		&mpu_config.mpu_regions[mpu_config.sram_region]);
+
+	u32_t mpu_reg_index = static_regions_num;
+
+	/* In NXP MPU architecture the dynamic regions are
+	 * programmed on top of existing SRAM region configuration.
+	 */
+
+	mpu_reg_index = _mpu_configure_regions(dynamic_regions,
+		regions_num, mpu_reg_index, false);
+
+	if (mpu_reg_index != -EINVAL) {
+
+		/* Disable the non-programmed MPU regions. */
+		for (int i = mpu_reg_index; i < _get_num_regions(); i++) {
+
+			LOG_DBG("disable region 0x%x", i);
+			/* Disable region */
+			SYSMPU->WORD[i][0] = 0;
+			SYSMPU->WORD[i][1] = 0;
+			SYSMPU->WORD[i][2] = 0;
+			SYSMPU->WORD[i][3] = 0;
+		}
+	}
+
+	return mpu_reg_index;
+}
+
+#if 0
```

Comment:
just delete this #if 0 code if we are not using it

Reply:
Thanks, this was forgotten. Removed now.

Reply:
(addressed)